From 1dcbbae395014efd6dbd8463cac3f33fe0191719 Mon Sep 17 00:00:00 2001
From: Will Thompson <wjt@endlessos.org>
Date: Fri, 29 Apr 2022 11:18:36 +0100
Subject: [PATCH 07/13] metainfo: Mark for translation

The information in the metainfo file is displayed to users in software
centres such as GNOME Software, and so should be translated.

Fixes #194
---
 .gitignore                                    |  1 +
 Makefile.am                                   |  3 +-
 ....metainfo.xml => gobby-0.5.metainfo.xml.in | 30 +++++++++----------
 po/POTFILES.in                                |  1 +
 update-potfiles                               |  1 +
 5 files changed, 20 insertions(+), 16 deletions(-)
 rename gobby-0.5.metainfo.xml => gobby-0.5.metainfo.xml.in (52%)

diff --git a/.gitignore b/.gitignore
index 6a40a1c..fcbd252 100644
--- a/.gitignore
+++ b/.gitignore
@@ -11,6 +11,7 @@ depcomp
 .dirstamp
 gobby-0.5
 gobby-0.5.desktop
+gobby-0.5.metainfo.xml
 gobby.iss
 *.gschema.valid
 *.gschema.xml
diff --git a/Makefile.am b/Makefile.am
index 3c57e64..aac2bb7 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -67,11 +67,12 @@ clean-local:
 	-rm -rf gobby-0.5.desktop
 
 @INTLTOOL_DESKTOP_RULE@
+@INTLTOOL_XML_RULE@
 
 EXTRA_DIST += README.md
 
 # For the manual
-EXTRA_DIST += gobby-0.5.desktop.in gobby-0.5.metainfo.xml
+EXTRA_DIST += gobby-0.5.desktop.in gobby-0.5.metainfo.xml.in
 DISTCLEANFILES += gnome-doc-utils.make
 DISTCHECK_CONFIGURE_FLAGS = --disable-scrollkeeper
 
diff --git a/gobby-0.5.metainfo.xml b/gobby-0.5.metainfo.xml.in
similarity index 52%
rename from gobby-0.5.metainfo.xml
rename to gobby-0.5.metainfo.xml.in
index 8d989f7..2eda036 100644
--- a/gobby-0.5.metainfo.xml
+++ b/gobby-0.5.metainfo.xml.in
@@ -7,17 +7,17 @@
   <provides>
     <binary>gobby-0.5</binary>
   </provides>
-  <name>Gobby</name>
-  <summary>Collaborative text editor</summary>
+  <_name>Gobby</_name>
+  <_summary>Collaborative text editor</_summary>
 
   <description>
-    <p>
+    <_p>
       Gobby is a free collaborative editor supporting multiple documents in one session and a multi-user chat.
       It runs on Microsoft Windows, Mac OS X, Linux and other Unix-like platforms.
-    </p>
-    <p>
+    </_p>
+    <_p>
       It uses GTK+ as its windowing toolkit and thus integrates nicely into the GNOME desktop environment.
-    </p>
+    </_p>
   </description>
 
   <screenshots>
@@ -33,15 +33,15 @@
     <release version="0.6" date="2021-01-31">
       <description>
         <ul>
-          <li>Remove support for GTK+ 2.x; at least GTK+ 3.6 is required now.</li>
-          <li>Enable TCP keepalives, so that inactive connections will drop automatically after a short time.</li>
-          <li>Gobby is now registered as a handler for infinote:// URLs.</li>
-          <li>Remember hosts previously connected to and presents them in the list next time Gobby is started until expilictly removed.</li>
-          <li>Gobby can now show connection parameters, in particular TLS parameters used for the connection.</li>
-          <li>Gobby's configuration options are stored in GSettings instead of a custam config.xml file.</li>
-          <li>The option to not set a certificate in the initial welcome dialog is no longer available.</li>
-          <li>Gobby can now use the default system CAs on Windows.</li>
-          <li>Small bug fixes and code base modernization.</li>
+          <_li>Remove support for GTK+ 2.x; at least GTK+ 3.6 is required now.</_li>
+          <_li>Enable TCP keepalives, so that inactive connections will drop automatically after a short time.</_li>
+          <_li>Gobby is now registered as a handler for infinote:// URLs.</_li>
+          <_li>Remember hosts previously connected to and presents them in the list next time Gobby is started until expilictly removed.</_li>
+          <_li>Gobby can now show connection parameters, in particular TLS parameters used for the connection.</_li>
+          <_li>Gobby's configuration options are stored in GSettings instead of a custam config.xml file.</_li>
+          <_li>The option to not set a certificate in the initial welcome dialog is no longer available.</_li>
+          <_li>Gobby can now use the default system CAs on Windows.</_li>
+          <_li>Small bug fixes and code base modernization.</_li>
         </ul>
       </description>
     </release>
diff --git a/po/POTFILES.in b/po/POTFILES.in
index dca2f6c..09fe615 100644
--- a/po/POTFILES.in
+++ b/po/POTFILES.in
@@ -58,3 +58,4 @@ code/util/i18n.hpp
 code/util/uri.cpp
 code/window.cpp
 gobby-0.5.desktop.in
+gobby-0.5.metainfo.xml.in
diff --git a/update-potfiles b/update-potfiles
index e8fd095..43ff905 100755
--- a/update-potfiles
+++ b/update-potfiles
@@ -8,3 +8,4 @@ done
 grep --binary-files=without-match "_(" code -R | \
 	sed -e 's/:.*//' | uniq | sort > po/POTFILES.in
 echo gobby-0.5.desktop.in >> po/POTFILES.in
+echo gobby-0.5.metainfo.xml.in >> po.POTFILES.in
-- 
2.30.2

